server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -75,12 +79,15 @@
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
         Preconditions.checkArgument(request instanceof GetMailboxesRequest);
         GetMailboxesRequest mailboxesRequest = (GetMailboxesRequest) request;
-        return Stream.of(
+        TimeMetric timeMetric = metricFactory.timer(METHOD_NAME.getName());
+        Stream<JmapResponse> responses = Stream.of(
                 JmapResponse.builder().clientId(clientId)
                 .response(getMailboxesResponse(mailboxesRequest, mailboxSession))
                 .properties(mailboxesRequest.getProperties().map(this::ensureContainsId))
                 .responseName(RESPONSE_NAME)
                 .build());
+        timeMetric.elapseTimeInMs();
+        return responses;
```

Comment:
Ok, i didn't understand your question.
So the response is that i prefere to handle metrics at lower level, where metric name really represents what it means; not what it has done before doing what it's expected.
We may introduce in the future a more global metric on request handling.